src/Window.cpp

```diff
@@ -331,6 +333,29 @@ void Gosu::Window::button_down(Button button)
     }
 }
 
+Gosu::Bitmap Gosu::Window::to_bitmap()
+{
+    ensure_current_context();
+
+    std::vector<Color> pixels(width()*height());
+    std::vector<Color::Channel> channels(width()*height()*4);
```

Comment:
From a performance POV it would be nicer to create the Bitmap first, then `glReadPixels` into the bitmap (it's designed to be compatible with GL_RGBA, see https://github.com/gosu/gosu/blob/master/src/TexChunk.cpp#L118), and then fix the bitmap's orientation by swapping each row with memcpy and a temporary buffer or plain XOR.

Reply:
I tried memcpy in my first few test and always got segmentation faults but might give it a second shot now that I know that glReadPixels is actually working and returning the right values.

Reply:
Looked again at TexChunk and the docu. The problem is, that glReadPixels doesn't return a single uint32 containing the 4 channels but instead giving me 4 single bytes which i have to combine to a color first before assigning it to the bitmap. Would be cool to have something like Array.pack in Ruby to just squash them together or better making glReadPixels actually return the pixels as one value.